pkg/activator/net/helpers.go

```diff
@@ -0,0 +1,53 @@
+package activator
+
+import (
+	"net"
+	"strconv"
+
+	corev1 "k8s.io/api/core/v1"
+	"knative.dev/serving/pkg/apis/networking"
+)
+
+// EndpointsToDests takes an endpoints object and a port name and returns a list
+// of l4 dests in the endpoints object which have that port
+func EndpointsToDests(endpoints *corev1.Endpoints) []string {
+	ret := []string{}
+
+	for _, es := range endpoints.Subsets {
+		portVal := int32(0)
+		for _, port := range es.Ports {
+			if port.Name == probePortName {
+				portVal = port.Port
+				break
+			}
+		}
+
+		// This endpoint doesnt have a port we want to probe
+		if portVal == 0 {
+			continue
+		}
+
+		pvs := strconv.Itoa(int(portVal))
+		for _, addr := range es.Addresses {
```

Comment:
You could nest this loop to line 20 to avoid the `continue` on line 27, also could do `pvs := strconv.Itoa(port.Port)` instead of assigning to `portVal`.